{
  "version": "2.1.0",
  "$schema": "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json",
  "runs": [
    {
      "tool": {
        "driver": {
          "fullName": "Docker Scout",
          "informationUri": "https://docker.com/products/docker-scout",
          "name": "docker scout",
          "rules": [
            {
              "id": "CVE-2018-20225",
              "name": "OsPackageVulnerability",
              "shortDescription": {
                "text": "CVE-2018-20225: OWASP Top Ten 2017 Category A9 - Using Components with Known Vulnerabilities"
              },
              "helpUri": "https://scout.docker.com/v/CVE-2018-20225?s=gitlab&n=pip&t=pypi&vr=%3E%3D0",
              "help": {
                "text": "An issue was discovered in pip (all versions) because it installs the version with the highest version number, even if the user had intended to obtain a private package from a private index. This only affects use of the `--extra-index-url` option, and exploitation requires that the package does not already exist in the public index (and thus the attacker can put the package there with an arbitrary version number).\n",
                "markdown": "> An issue was discovered in pip (all versions) because it installs the version with the highest version number, even if the user had intended to obtain a private package from a private index. This only affects use of the `--extra-index-url` option, and exploitation requires that the package does not already exist in the public index (and thus the attacker can put the package there with an arbitrary version number).\n\n|                |                                              |\n|----------------|----------------------------------------------|\n| Package        | pkg:pypi/pip@24.0                            |\n| Affected range | >=0                                          |\n| Fixed version  | not fixed                                    |\n| CVSS Score     |                                          7.8 |\n| CVSS Vector    | CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H |\n"
              },
              "properties": {
                "affected_version": ">=0",
                "cvssV3": 7.8,
                "cvssV3_severity": "HIGH",
                "cvssV3_vector": "CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
                "fixed_version": "not fixed",
                "purls": [
                  "pkg:pypi/pip@24.0"
                ],
                "security-severity": "7.8",
                "tags": [
                  "HIGH"
                ]
              }
            }
          ],
          "version": "1.8.0"
        }
      },
      "results": [
        {
          "ruleId": "CVE-2018-20225",
          "ruleIndex": 0,
          "kind": "fail",
          "level": "error",
          "message": {
            "text": "  Vulnerability   : CVE-2018-20225                                \n  Severity        : HIGH                                          \n  Package         : pkg:pypi/pip@24.0                             \n  Affected range  : >=0                                           \n  Fixed version   : not fixed                                     \n  CVSS Score      : 7.8                                           \n  CVSS Vector     : CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H  \n  EPSS Score      : 0.001120                                      \n  EPSS Percentile : 0.446160                                      \n"
          },
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "/usr/local/lib/python3.12/site-packages/pip-24.0.dist-info/METADATA"
                }
              }
            },
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "/usr/local/lib/python3.12/site-packages/pip-24.0.dist-info/RECORD"
                }
              }
            },
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "/usr/local/lib/python3.12/site-packages/pip-24.0.dist-info/top_level.txt"
                }
              }
            }
          ]
        }
      ]
    }
  ]
}
